sound/soc/sof/sof-priv.h

```diff
@@ -402,10 +400,13 @@ struct snd_sof_dev {
 	struct snd_soc_component_driver plat_drv;
 
 	/* power states related */
-	enum sof_d0_substate d0_substate;
+	enum sof_dsp_power_state dsp_power_state;
 	/* flag to track if the intended power target of suspend is S0ix */
 	bool s0_suspend;
 
+	/* number of pipelines running when the DSP is in D0i3 */
+	u32 D0i3_pipeline_count;
```

Comment:
pipeline or stream? d0i3-compatible or anyone? make it clear in the comment.

Reply:
"number of pipelines running when the DSP is in D0i3". Isnt it clear enough that only D0i3-compatible ones can run in D0i3?

Reply:
I specifically chose the word pipeline because the streams are suspended from ALSA point of view. Only the pipelines are running in the DSP

Reply:
> I specifically chose the word pipeline because the streams are suspended from ALSA point of view. Only the pipelines are running in the DSP

But the count here is of stream, e.g. though KWD stream includes 2 pipelines, we only increase the count by one here.

Reply:
Yes, but that's because the KWD pipeline does not contain a host PCM and it never receives the suspend trigger.